os/arch/arm/src/amebasmart/amebasmart_i2c.c

```diff
@@ -1284,6 +1284,7 @@ static uint32_t rtk_i2c_resume(uint32_t expected_idle_time, void *param)
 #ifdef CONFIG_PM
 void i2c_pminitialize(void)
 {
+	(void)pm_domain_register("I2C");
```

Comment:
I think enum is better option instead of string. then you can use like below.

```
enum pm_domain_e
{
	PM_DOMAIN_I2C = 0,
	PM_DOMAIN_I2S,
	...
	PM_DOMAIN_MAX,
};

int pm_domain_register(char *domain) {
	...
	for (index = 0; index < PM_DOMAIN_MAX; index++) {
		...
	}
	...
}
```
Then you should not define CONFIG_PM_NDOMAINS.
Is there any specific reason for string value?

Reply:
We don't know which app to use PM.
And there was a request that it was not right to change the enum provided by the interface frequently.

We are going to operate both static and dynamic tables in the future for performance
It seems necessary to modify the name of CONFIG_PM_NDOMAINS.
This is simply a static table size.